jep/223/README.adoc

```diff
@@ -269,6 +269,108 @@ The intent of this proposal is to improve overall security for Jenkins instances
 All configuration items that require `Jenkins.ADMINISTER`, `Jenkins.RUN_SCRIPTS`, `PluginManager.CONFIGURE_UPDATECENTER`, and `PluginManager.UPLOAD_PLUGINS`  are being reviewed to determine if they can be changed to require the new `Jenkins.MANAGE` permission type.
 The greatest risk is that some configuration may be _more restrictive_ than necessary, either because it was overlooked or due to differences in opinion during the review process.
 
+=== Specific focus on ManagementLink.getRequiredPermission()
+
+Previously, the “/manage” page was only accessible to a user with the `Jenkins.ADMINISTER` permission, and the `ManagementLink` links on this page were protected by virtue of the StaplerProxy performing a permission check for `Jenkins.ADMINISTER`.  Now that the “/manage” page can be accessible by other permission types it is important that all classes that extend `ManagementLink` return a sensible default access permission.
+
+To do it we need to change the default value of getRequiredPermission() from:
+
+```
+public @CheckForNull Permission getRequiredPermission() {
+    return null;
+}
+```
+
+to default `Jenkins.ADMINISTER`:
+```
+public @CheckForNull Permission getRequiredPermission() {
+    return Jenkins.ADMINISTER;
+}
+```
+This change  only removes the links that are displayed in the "/manage" page for users that don't have the related permission.  Users with the `Jenkins.ADMINISTER` permission will not see any changes in the “/manage” page, and users with the `Jenkins.MANAGE` permission will only see links that have explicitly override the default implementation of `getRequiredPermission`.
+As this could break plugin compatibility, we did an analysis on the plugin usage of `getRequiredPermission` or `requiredPermission` in jelly/groovy.
+
+.Table Plugin usage analysis of ManagementLink.getRequiredPermission()
+|===
+|Plugin |Analyse |Reason
+
+|multi-slave-config-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|scriptler-plugin
+|No change
+|Manages its own permission in the ManagementLink
+
+|p4-plugin
+|No change
+|Based on scriptler-plugin
+
+|periodic-jobs-balancer-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|gerrit-trigger-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|lenient-shutdown-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|script-security-plugin
+|No change
+|getRequiredPermission() is overrided with `RUN_SCRIPT` in the ManagementLink
+
+|active-choices-plugin
+|No change
+|Based on scriptler-plugin
+
+|docker-plugin
+|Change
+|Does not override getRequiredPermission() in the ManagementLink and use it in jelly. The management page will be restricted to `ADMINISTER`.
```

Comment:
Due to [DockerManagement#getTarget](https://github.com/jenkinsci/docker-plugin/blob/master/src/main/java/com/nirima/jenkins/plugins/docker/DockerManagement.java#L82-L85), requiring admin is the expected behavior.

Reply:
When is the getTarget used in the lifecycle of the management page ? Before displaying the page or on save ?

Reply:
If you see a class in Jenkins ecosystem, implementing StaplerProxy with its "getTarget" method, this is used during Stapler routing process. When the routing arrives on the class (as a fragment/step inside the "URL path" traversal), it will first check the getTarget() and use the return of this method. This could be used to check permission (throwing AccessDeniedException) or changing the target for this fragment.